includes/class-post-types.php

```diff
@@ -0,0 +1,548 @@
+<?php
+/**
+ * Post Types class for consolidating all custom post type and related meta registrations.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Activity\Activity;
+use Activitypub\Collection\Extra_Fields;
+use Activitypub\Collection\Followers;
+use Activitypub\Collection\Inbox;
+use Activitypub\Collection\Outbox;
+use Activitypub\Collection\Remote_Actors;
+
+/**
+ * Post Types class.
+ */
+class Post_Types {
+	/**
+	 * Initialize the class, registering all custom post types and post meta.
+	 */
+	public static function init() {
+		self::register_remote_actors_post_type();
+		self::register_inbox_post_type();
+		self::register_outbox_post_type();
+		self::register_extra_fields_post_types();
+		self::register_activitypub_post_meta();
+
+		\add_action( 'rest_api_init', array( self::class, 'register_ap_actor_rest_field' ) );
+
+		\add_filter( 'activitypub_get_actor_extra_fields', array( Extra_Fields::class, 'default_actor_extra_fields' ), 10, 2 );
```

Comment:
this feels a bit like a hack, we did because we wanted to keep the collections free from `init` hooks!

What do you think about also move the `default_actor_extra_fields` function to this class?

Reply:
As an alternative, how about we call `default_actor_extra_fields()` from `get_actor_fields()` when they've not been created yet? That would also avoid the side effect of creating posts in a filter callback. Happy to do that in a follow-up.